sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ImmutableStorageWithVersioningTest.groovy

```diff
@@ -130,36 +138,37 @@ class ImmutableStorageWithVersioningTest extends APISpec {
                 .endpoint(env.versionedAccount.blobEndpoint)
                 .buildClient()
 
-            def options = new ListBlobContainersOptions().setPrefix(namer.getResourcePrefix())
-            for (def container : cleanupClient.listBlobContainers(options, null)) {
-                def containerClient = cleanupClient.getBlobContainerClient(container.getName())
+            def containerClient = cleanupClient.getBlobContainerClient(vlwContainerName)
+            def containerProperties = containerClient.getProperties()
 
-                if (container.getProperties().getLeaseState() == LeaseStateType.LEASED) {
-                    createLeaseClient(containerClient).breakLeaseWithResponse(new BlobBreakLeaseOptions().setBreakPeriod(Duration.ofSeconds(0)), null, null)
-                }
-                if (container.getProperties().isImmutableStorageWithVersioningEnabled()) {
-                    options = new ListBlobsOptions().setDetails(new BlobListDetails().setRetrieveImmutabilityPolicy(true).setRetrieveLegalHold(true))
-                    for (def blob: containerClient.listBlobs(options, null)) {
-                        def blobClient = containerClient.getBlobClient(blob.getName())
-                        if (blob.getProperties().hasLegalHold()) {
-                            blobClient.setLegalHold(false)
-                        }
-                        if (blob.getProperties().getImmutabilityPolicy().getPolicyMode() != null) {
-                            sleepIfRecord(5 * 1000)
-                            blobClient.deleteImmutabilityPolicy()
-                        }
-                        blobClient.delete()
+            if (containerProperties.getLeaseState() == LeaseStateType.LEASED) {
+                createLeaseClient(containerClient).breakLeaseWithResponse(new BlobBreakLeaseOptions().setBreakPeriod(Duration.ofSeconds(0)), null, null)
+            }
+            if (containerProperties.isImmutableStorageWithVersioningEnabled()) {
+                def options = new ListBlobsOptions().setDetails(new BlobListDetails().setRetrieveImmutabilityPolicy(true).setRetrieveLegalHold(true))
+                for (def blob: containerClient.listBlobs(options, null)) {
+                    def blobClient = containerClient.getBlobClient(blob.getName())
+                    if (blob.getProperties().hasLegalHold()) {
+                        blobClient.setLegalHold(false)
+                    }
+                    if (blob.getProperties().getImmutabilityPolicy().getPolicyMode() != null) {
+                        sleepIfRecord(5 * 1000)
```

Comment:
Is this sleep waiting for the legal status or immutable policy change to propagate on the service?

Reply:
no, that's totally unnecessary, I will remove this.